### PR TITLE
Update hostsubnets on vSphere

### PR DESCRIPTION
**Pull request by denisj07**
This PR adds vSphere to the baremetal if statement.
It's tested on our installer-provisioned 4.5 cluster (https://docs.openshift.com/container-platform/4.5/installing/installing_vsphere/installing-vsphere-installer-provisioned.html), and it seems like nothing else is needed in order to run the egressip-ipam-operator on vSphere.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -8,13 +8,13 @@ require (
...
 	github.com/openshift/cloud-credential-operator v0.0.0-20200303191255-cc686af9c1b9
-	github.com/openshift/machine-api-operator v0.2.1-0.20200402110321-4f3602b96da3
+	github.com/openshift/machine-api-operator v0.0.0-20200514040855-461bde5e8c5e
 	github.com/operator-framework/operator-sdk v0.14.0
 	github.com/redhat-cop/operator-utils v0.1.1
 	github.com/scylladb/go-set v1.0.2
 	github.com/spf13/pflag v1.0.5
-	k8s.io/api v0.18.0-rc.1
-	k8s.io/apimachinery v0.18.0-rc.1
+	k8s.io/api v0.18.2
```
Comment (raffaelespazzoli) [inline go.mod:16]: why this changes? these versions are based on the operator-sdk dependencies and as far as I understand should not be changed.

**Review by raffaelespazzoli — CHANGES_REQUESTED** [review]
I'm good with the code change. If vsphere does not have the concept of regions (can you confirm that?), then yes it will simply behave like bare metal.
Can you also add a section in the doc about vSphere?

File: go.mod (hunk 1 above)
Comment (denisj07) [reply go.mod:16]: That's a good question, I have to confess that I haven't done that much with golang so far. Whenever I change those numbers back to the old version, v0.18.0-rc.1, `go mod download` changes the version to v0.18.2.

Is it possible that my change of the machine-api-operator to a newer revision (https://github.com/openshift/machine-api-operator/commit/461bde5e8c5e175639187ed78571fdba36afc1ac, had to update this because of https://github.com/openshift/library-go/commit/c1de486c7d47605416673079735c41703b03b7b4, the bitbucket repo is not available anymore) automatically switches k8s.io/api and k8s.io/apimachinery to v0.18.2 because they're using that version, too (see https://github.com/openshift/machine-api-operator/commit/461bde5e8c5e175639187ed78571fdba36afc1ac#diff-37aff102a57d3d7b797f152915a6dc16L30-R33)?

**Comment by denisj07** [comment]
vSphere does not really support the concept of regions in the same way that AWS does. The OpenShift installer can only connect to a single datacenter (https://docs.openshift.com/container-platform/4.5/installing/installing_vsphere/installing-vsphere-installer-provisioned.html#prerequisites). Theoretically the servers within that datacenter could be in different physical regions, but in this case personally I would create another MachineSet and let the operator (I think it's the machine-config operator) label that nodes accordingly. These labels can then be used within the EgressIPAM object to select the correct nodes.

**Comment by raffaelespazzoli** [comment]
@denisj07 the code looks good to me, however I don't have a way to actually test this. How did you test this code?

**Comment by raffaelespazzoli** [comment]
@denisj07 I tested this and it works fine. We need just one correction to the doc: the syntax for nodeSelector is wrong. Can you add the `matchLabel` field like for the AWS example. And can you add it also in the example at the beginning of the doc?

**Comment by raffaelespazzoli** [comment]
@denisj07 one more thing, it seems to me that RHV will behave like vSphere. Do you mind adding it to the `if` condition and also adding some doc?

**Comment by denisj07** [comment]
@raffaelespazzoli sorry for not replying back, I was busy with some other projects in the last few days...
Regarding your question: I've tested this code manually on our own vSphere installation as I have no experience in writing tests for Operators.

What exactly do you mean by
> We need just one correction to the doc: the syntax for nodeSelector is wrong.

My worker nodes were automatically provisioned with exactly the same label that's already published in the documentation: 
```yaml
kind: Node
apiVersion: v1
metadata:
  labels:
    beta.kubernetes.io/arch: amd64
    beta.kubernetes.io/os: linux
    egressGateway: 'true'
    kubernetes.io/arch: amd64
    kubernetes.io/hostname: my-fancy-hostname
    kubernetes.io/os: linux
    node-role.kubernetes.io/worker: '' <-------
    node.openshift.io/os_id: rhcos
```
There was nothing I had to change on my side - so the documentation seems to be correct already.

Regarding RHV, I could absolutely add it to the code, but I would create a new PR for that...

**Comment by raffaelespazzoli** [comment]
@denisj07 the node selector must look like this:
```yaml
  nodeSelector:
    matchLabels:
      node-role.kubernetes.io/worker: ""
```
I'm ok with bundling RHV and VmWare in the same PR, but if you prefer doing two of them, that is also fine by me.

**Comment by denisj07** [comment]
@raffaelespazzoli you are absolutely right! Now that I see it I remember going through the crd definitions trying to figure out why the nodeSelector was not working :'D
